dotnet/src/Connectors/Connectors.AzureOpenAI/AzureOpenAIPromptExecutionSettings.cs

```diff
@@ -0,0 +1,432 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Azure.AI.OpenAI;
+using Microsoft.SemanticKernel.ChatCompletion;
+using Microsoft.SemanticKernel.Text;
+
+namespace Microsoft.SemanticKernel.Connectors.AzureOpenAI;
+
+/// <summary>
+/// Execution settings for an AzureOpenAI completion request.
+/// </summary>
+[JsonNumberHandling(JsonNumberHandling.AllowReadingFromString)]
+public sealed class AzureOpenAIPromptExecutionSettings : PromptExecutionSettings
```

Comment:
We need to think about whether or not we depend on the OpenAI connector and extend the `OpenAIPromptExecutionSettings`.

Reply:
I considered this at first, but not sure doing so will polute all the OpenAI specific services, extensions. If we can be more segregated in this aspect I would go for it, IMHO.

Reply:
I would avoid such dependency and keep these settings separately in order to be able to add more functionality to them in the future independently.

Reply:
I would avoid replicating the inheritance structure of the OpenAI and AzureOpenAI SDKs in the public classes of SK to prevent unnecessary breaking changes if the AzureOpenAI SDK stops the inheritance for any reason. Code duplication will be the price for flexibility and the absence of breaking changes.

Reply:
Resolving the comment for now. The final decision will be made in scope of the issue - https://github.com/microsoft/semantic-kernel/issues/6915